File: test/data/javascript/simple.js
// Simple JavaScript test file

function hello(name) {
    console.log(`Hello, ${name}!`);
}

class Calculator {
    constructor() {
        this.result = 0;
    }
    
    add(a, b) {
        return a + b;
    }
    
    subtract(a, b) {
        return a - b;
    }
}

const calc = new Calculator();
const sum = calc.add(5, 3);

// Arrow function
const multiply = (x, y) => x * y;

// Object literal
const person = {
    name: "John",
    age: 30,
    greet: function() {
        return `Hi, I'm ${this.name}`;
    }
};

// Array methods
const numbers = [1, 2, 3, 4, 5];
const doubled = numbers.map(n => n * 2);

// Async function
async function fetchData(url) {
    try {
        const response = await fetch(url);
        return await response.json();
    } catch (error) {
        console.error('Error:', error);
    }
}

// Export
export { hello, Calculator, multiply };

File: test/data/kotlin/simple.kt
// Simple Kotlin example demonstrating modern language features
data class Person(val name: String, val age: Int)

fun main() {
    val greeting = "Hello, Kotlin!"
    println(greeting)
    
    // Lambda expressions with semantic refinements
    val numbers = listOf(1, 2, 3, 4, 5)
    val doubled = numbers.map { it * 2 }
    
    // Higher-order functions
    val transform: (Int) -> Int = { x -> x * x }
    val squared = numbers.map(transform)
    
    // Immutable vs mutable variables
    val immutable = "cannot change"
    var mutable = "can change"
    mutable = "changed"
    
    // Constructor calls with refinements
    val person = Person("Alice", 30)
    
    // When expression (pattern matching)
    when (person.age) {
        in 0..12 -> println("Child")
        in 13..19 -> println("Teenager")
        else -> println("Adult")
    }
    
    // Nullable types
    val nullable: String? = null
    val length = nullable?.length ?: 0
    
    // Extension functions
    fun String.isPalindrome(): Boolean {
        return this == this.reversed()
    }
    
    println("racecar".isPalindrome())
}

// Class with different function types
class Calculator {
    // Regular method
    fun add(a: Int, b: Int): Int = a + b
    
    // Property with getter/setter
    var result: Int = 0
        set(value) {
            field = if (value >= 0) value else 0
        }
    
    // Companion object (similar to static)
    companion object {
        const val PI = 3.14159
        fun createDefault() = Calculator()
    }
}

// Sealed class for algebraic data types
sealed class Result<T> {
    data class Success<T>(val value: T) : Result<T>()
    data class Error<T>(val message: String) : Result<T>()
}

// Async/coroutine function
suspend fun fetchData(): String {
    delay(1000)
    return "Data fetched"
}